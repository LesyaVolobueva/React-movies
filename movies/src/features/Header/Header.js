import React from 'react';
import styles from './header.module.scss';
import { connect } from 'react-redux';
import { logout } from '../Auth/authActions';
import PropTypes from 'prop-types';
import WithTranslation from '../WithTranslation/WithTranslation';


const mapStateToProps = (state) => ({
  user: state.auth.user,
});

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logout()),
});

export class Header extends React.Component {
  render() {
    const { logout, user, lang } = this.props;

    return (
      <div className={styles.div}>
        <h1 className={styles.title}>{lang.movies}</h1>
        {user &&
        <button
          className={styles.logout}
          onClick={logout}
        >
          {lang.logOut}
        </button>
        }
      </div>
    );
  }
}

Header.propTypes = {
  user: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      password: PropTypes.string,
    }),
  ]),
  logout: PropTypes.func,
  lang: PropTypes.objectOf(PropTypes.string),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(WithTranslation(Header));
