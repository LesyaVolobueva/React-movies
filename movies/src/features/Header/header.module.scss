.div {
  font-family: sans-serif;
  position: relative;
  border-bottom: 2px solid #000;
  padding: 20px;
}

.title {
  font-size: 24px;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: bold;
}

.logout {
  position: absolute;
  right: 80px;
  top: 10px;
  padding: 10px;
  font-size: 12px;
  border: 1px solid #abc2e8;
  background: #abc2e8;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  border-radius: 5px;
  cursor: pointer;
}
