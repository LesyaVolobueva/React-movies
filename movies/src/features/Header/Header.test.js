import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import { Header } from './Header';

const lang = {
  edit: 'Edit',
  delete: 'Delete',
  save: 'Save',
  back: 'Go Back',
  signUp: 'Sign Up',
  login: 'Log In',
  movies: 'Movies',
  logOut: 'Log Out',
  imgUrl: 'Img Url',
  title: 'Title',
  director: 'Director',
  genres: 'Genres',
  description: 'Description',
  likes: 'Likes',
  actors: 'Actors',
  searchPlaceholder: 'Enter the name',
  byLikes: 'By likes',
  byRate: 'By rate',
  pleaseLogin: 'Please log in',
  userLogin: 'Your login',
  userPassword: 'Your password',
  passwordIncorrect: 'Password is incorrect!',
  userDoNotExist: 'User doesn\'t exist',
  dontHaveAcc: 'Don\'t have an account?',
  pleaseSignUp: 'Please sign up',
  suchUserExists: 'User with such login exists!',
  alreadyHaveAcc: 'Already have the account?',
};

describe('-- Header --', () => {
  let mockStore;
  let store;

  beforeEach(() => {
    mockStore = configureStore();
    store = {
      lang: {
        lang: 'en',
      },
      movies: {
        actors: [],
        movies: [],
      },
      auth: {
        user: {},
        logout: jest.fn(),
      }
    };
  });


  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
            <Header
              lang={lang}
            />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Logout function is called on button click', () => {
    const mockedFunc = jest.fn();
    const props = {
      logout: jest.fn(),
      lang,
      user: {},
    };

    const wrapper = mount(
      <MemoryRouter>
        <Header {...props}/>
      </MemoryRouter>
    );

    expect(props.logout.mock.calls.length).toBe(0);

    wrapper.find('button').simulate('click');

    expect(props.logout.mock.calls.length).toBe(1);
  });
});
