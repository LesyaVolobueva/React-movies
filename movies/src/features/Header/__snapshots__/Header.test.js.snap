// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-- Header -- renders correctly 1`] = `
<div
  className={undefined}
>
  <h1
    className={undefined}
  >
    Movies
  </h1>
</div>
`;
